djconnectwise/models.py

```diff
@@ -948,8 +1356,20 @@ class Meta:
     def __str__(self):
         return self.name
 
+    def get_calendar(self, company_id):
+        try:
+            if self.calendar:
+                return self.calendar
+            elif self.based_on == 'Customer':
+                return Company.objects.get(id=company_id)
```

Comment:
This method should not return a Calendar in some circumstances and a Company in others. Please adjust so that it always returns a Calendar.

Reply:
Oh oops, that shouldnt have been returning a company in the first place